homework_10/FileLogger.java

```diff
@@ -0,0 +1,61 @@
+import enums.LoggingLevel;
+import exceptions.FileMaxSizeException;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class FileLogger {
+    private final FileLoggerConfiguration config;
+
+    public FileLogger(FileLoggerConfiguration config) {
+        this.config = config;
+    }
+
+    public void debug(String message) {
+        if (isDebugEnabled()) {
+            log(message, LoggingLevel.DEBUG);
+            log(message, LoggingLevel.INFO);
+        }
+    }
+
+    public void info(String message) {
+        if (isInfoEnabled()) {
+            log(message, LoggingLevel.INFO);
+        }
+    }
+
+    private void log(String message, LoggingLevel loggingLevel) {
+        File file = new File(config.getFileName());
+        if (isMaxFileSizeExceeded(file)) {
+            createNewLogFile(file.getName());
+        } else {
+            try (FileWriter writer = new FileWriter(file, true)) {
+                writer.write("[" + DateTimeUtils.getCurrentTimeAsString() + "][" + loggingLevel + "]" + " Message: " + message + "\n");
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private void createNewLogFile(String fileName) {
+        File newLogFile = new File(fileName);
+        try {
+            newLogFile.createNewFile();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private boolean isDebugEnabled() {
+        return config.getLoggingLevel() == LoggingLevel.DEBUG;
+    }
+
+    private boolean isInfoEnabled() {
+        return config.getLoggingLevel() == LoggingLevel.INFO;
```

Comment:
`INFO` is enabled when configuration level = `DEBUG` as  well.
Replace `return config.getLoggingLevel() == LoggingLevel.INFO;` with `return config.getLoggingLevel() == LoggingLevel.INFO || config.getLoggingLevel() == LoggingLevel.DEBUG;`, because when we have debug, then `INFO` should also work.